core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/RawlsApiService.scala

```diff
@@ -20,26 +21,28 @@ import scala.concurrent.ExecutionContext
-    Props(new RawlsApiServiceActor(workspaceServiceConstructor, userServiceConstructor, genomicsServiceConstructor, statisticsServiceConstructor, statusServiceConstructor, appVersion, googleClientId, submissionTimeout))
+  def props(workspaceServiceConstructor: UserInfo => WorkspaceService, userServiceConstructor: UserInfo => UserService, genomicsServiceConstructor: UserInfo => GenomicsService, statisticsServiceConstructor: UserInfo => StatisticsService, statusServiceConstructor: () => StatusService, appVersion: ApplicationVersion, googleClientId: String, submissionTimeout: FiniteDuration, rawlsMetricBaseName: String)(implicit executionContext: ExecutionContext): Props = {
+    Props(new RawlsApiServiceActor(workspaceServiceConstructor, userServiceConstructor, genomicsServiceConstructor, statisticsServiceConstructor, statusServiceConstructor, appVersion, googleClientId, submissionTimeout, rawlsMetricBaseName))
   }
 }
 
-class RawlsApiServiceActor(val workspaceServiceConstructor: UserInfo => WorkspaceService, val userServiceConstructor: UserInfo => UserService, val genomicsServiceConstructor: UserInfo => GenomicsService, val statisticsServiceConstructor: UserInfo => StatisticsService, val statusServiceConstructor: () => StatusService, val appVersion: ApplicationVersion, val googleClientId: String, val submissionTimeout: FiniteDuration)(implicit val executionContext: ExecutionContext) extends Actor
+class RawlsApiServiceActor(val workspaceServiceConstructor: UserInfo => WorkspaceService, val userServiceConstructor: UserInfo => UserService, val genomicsServiceConstructor: UserInfo => GenomicsService, val statisticsServiceConstructor: UserInfo => StatisticsService, val statusServiceConstructor: () => StatusService, val appVersion: ApplicationVersion, val googleClientId: String, val submissionTimeout: FiniteDuration, val rawlsMetricBaseName: String)(implicit val executionContext: ExecutionContext) extends Actor
   with RootRawlsApiService with WorkspaceApiService with EntityApiService with MethodConfigApiService with SubmissionApiService
   with AdminApiService with UserApiService with StandardUserInfoDirectives with BillingApiService with NotificationsApiService
-  with StatusApiService {
+  with StatusApiService with InstrumentationDirectives {
 
   def actorRefFactory = context
   def apiRoutes = options{ complete(OK) } ~ baseRoute ~ workspaceRoutes ~ entityRoutes ~ methodConfigRoutes ~ submissionRoutes ~ adminRoutes ~ userRoutes ~ billingRoutes ~ notificationsRoutes
   def registerRoutes = options{ complete(OK) } ~ createUserRoute ~ getUserStatusRoute
 
   def receive = runRoute(
-    swaggerRoute ~
-    versionRoute ~
-    statusRoute ~
-    pathPrefix("api") { apiRoutes } ~
-    pathPrefix("register") { registerRoutes }
+    instrumentRequest {
+      swaggerRoute ~
+        versionRoute ~
```

Comment:
Also should we instrument serving the swagger index and status checks? Those are likely to be pretty uninteresting metrics but I guess there is not much downside...